lib_eio/net.ml

```diff
@@ -143,13 +143,16 @@ module Sockaddr = struct
 
   type datagram = [
     | `Udp of Ipaddr.v4v6 * int
+    | `Unix_datagram of string
```

Comment:
I'm still digesting this diff, but not sure why Unix_datagram is named like this in the portable core.  We have unix domain sockets, and potentially Windows named pipes, so should this be a little more abstract?

Reply:
It could be. The unix(7) man-page says "The `AF_UNIX` (also known as `AF_LOCAL`) socket family ...", so perhaps `Local_datagram` would work?

I guess the question is whether it's useful to use Unix-domain sockets interchangeably with Windows named pipes. If so, they should share a name. If not, we can add another constructor for those.

Reply:
There's a somewhat orthogonal set of properties that each expose. Windows named pipes are unidirectional streams I think.  Is the common property here that some things have a local name, and some are datagrams or streams.

Reply:
According to https://learn.microsoft.com/en-us/windows/win32/ipc/pipe-names, a pipe name is in the format `\\ServerName\pipe\PipeName`. If a Linux process wanted to connect to a Windows named pipe on another machine, that would be quite different to connecting to a Unix socket with the same name, so I think they should be separate. Probably something like `Windows_pipe (server, local_name)`.

We could combine `Unix` and `Unix_datagram` I think. The only thing that distinguishes between them is `getnameinfo` to set the `NI_DGRAM` flag, but that isn't useful for Unix-domain sockets anyway.

I wonder if we should separate out Linux abstract socket addresses too. `Unix "/tmp/sock"` refers to a file, but `Unix "\000/tmp/sock"` doesn't. Or perhaps we shouldn't have a `Unix` network address at all, and instead add `Path.bind` and `Path.connect`. They are part of the mount namespace rather than the network namespace.